### PR TITLE
Fixes #7856: Use a mustache template from a string

### PR DESCRIPTION
**Pull request by peckpeck**
https://www.rudder-project.org/redmine/issues/7856

Replacing previous PR: https://github.com/Normation/ncf/pull/315

### CODE REVIEW

**Comment by peckpeck** [comment]
PR updated

File: tree/30_generic_methods/file_from_string_mustache.cf [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+# @agent_version >=3.7
+#
+# @parameter template        String containing a template to be expanded
+# @parameter destination     Destination file
+#
+# @class_prefix file_from_string_mustache
+# @class_parameter destination
+# This bundle will define a class file_from_string_mustache_${destination}_{kept,repaired,not_ok,ok,reached}
+
+# This has been edited by rudder-dev, do not blindly commit but read git diff, check that class_prefix vs old_class_prefix are properly used,
```
Comment (jooooooon) [inline tree/30_generic_methods/file_from_string_mustache.cf]: You probably wanna remove this line :D

**Comment by jooooooon** [comment]
ping @peckpeck

File: tree/30_generic_methods/file_from_string_mustache.cf [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "class_prefix"      string => canonify(join("_", "promisers"));
+    "args"               slist => { "${template}", "${destination}" };
+    "content"           string => string_mustache("${template}", datastate());
+
+  files:
+    "${destination}"
+      create        => "true",
+      edit_line     => insert_lines("${content}"),
+      edit_defaults => ncf_empty_select("true"),
+      classes     => classes_generic_two("${old_class_prefix}", "${class_prefix}");
```
Comment (ncharles) [inline tree/30_generic_methods/file_from_string_mustache.cf]: Correct indent there

**Comment by peckpeck** [comment]
Superseded by https://github.com/Normation/ncf/pull/340

**Pull request closed without merging**